### PR TITLE
Ctda 1806: audit large messages

### PR DESCRIPTION
**Pull request by liandra-giacomine**

### CODE REVIEW

File: app/controllers/DeparturesController.scala [hunk 1]
```diff
@@ -89,8 +89,13 @@ class DeparturesController @Inject()(
                         case submissionFailureRejected: SubmissionProcessingResult.SubmissionFailureRejected =>
                           BadRequest(submissionFailureRejected.responseBody)
                         case SubmissionProcessingResult.SubmissionSuccess =>
-                          auditService.auditEvent(DepartureDeclarationSubmitted, request.enrolmentId, departure.messages.head, request.channel)
-                          auditService.auditEvent(MesSenMES3Added, request.enrolmentId, departure.messages.head, request.channel)
+                          val len = request.headers.get(play.api.http.HeaderNames.CONTENT_LENGTH).get.toInt
```
Comment (RobertLaverick) [inline app/controllers/DeparturesController.scala:92]: we can probably move this into our request case class to make it a little cleaner to read here.

File: it/repositories/DepartureRepositorySpec.scala [hunk 2]
```diff
@@ -200,9 +200,13 @@ class DepartureRepositorySpec
```
Comment (RobertLaverick) [inline it/repositories/DepartureRepositorySpec.scala:200]: can you expand on why this has changed?

**Review by RobertLaverick — CHANGES_REQUESTED** [review]

File: it/repositories/DepartureRepositorySpec.scala (hunk 2 above)
Comment (liandra-giacomine) [reply it/repositories/DepartureRepositorySpec.scala:200]: @RobertLaverick when I ran integration tests I saw that there were failures for the ones I modified due to the datetime comparison being different in precision - I modified the precision for the match to work and the test to pass...happy to revert this change if you don't think it's necessary though

File: app/audit/DeclarationDetails.scala [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      _ => Constants.NewEnrolmentKey,
+      (_, _) => Constants.NewEnrolmentKey
+    )
+
+  def statistics: JsObject = Json.obj(
+    "itemCount"   -> (message \\ "TotNumOfIteHEA305").text.toInt,
+    "requestSize" -> requestSize
+  )
+
+  def declaration: JsObject =
```
Comment (drnaylor-oc) [inline app/audit/DeclarationDetails.scala:57]: def -> lazy val

**Review by drnaylor-oc — APPROVED** [review]
Looks good to me, @RobertLaverick, any last thoughts or shall we call this good?

**Review by RobertLaverick — APPROVED** [review]

**Pull request merged**